src/model/PythonPluginSearchPaths.hpp

```diff
@@ -0,0 +1,125 @@
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#ifndef MODEL_PYTHONPLUGINSEARCHPATHS_HPP
+#define MODEL_PYTHONPLUGINSEARCHPATHS_HPP
+
+#include <model/ModelAPI.hpp>
+#include "ModelObject.hpp"
+
+#include "../utilities/core/Filesystem.hpp"
+
+namespace openstudio {
+namespace model {
+
+  namespace detail {
+
+    class PythonPluginSearchPaths_Impl;
+
+  }  // namespace detail
+
+  /** PythonPluginSearchPaths is a ModelObject that wraps the OpenStudio IDD object 'OS:PythonPlugin:SearchPaths'. */
+  class MODEL_API PythonPluginSearchPaths : public ModelObject
+  {
+   public:
+    /** @name Constructors and Destructors */
+    //@{
+
+    virtual ~PythonPluginSearchPaths() override = default;
+    // Default the copy and move operators because the virtual dtor is explicit
+    PythonPluginSearchPaths(const PythonPluginSearchPaths& other) = default;
+    PythonPluginSearchPaths(PythonPluginSearchPaths&& other) = default;
+    PythonPluginSearchPaths& operator=(const PythonPluginSearchPaths&) = default;
+    PythonPluginSearchPaths& operator=(PythonPluginSearchPaths&&) = default;
+
+    //@}
+
+    static IddObjectType iddObjectType();
+
+    /** @name Getters */
+    //@{
+
+    bool addCurrentWorkingDirectorytoSearchPath() const;
+
+    bool addInputFileDirectorytoSearchPath() const;
+
+    bool addepinEnvironmentVariabletoSearchPath() const;
+
+    std::vector<openstudio::path> searchPaths() const;
+
+    //@}
+    /** @name Setters */
+    //@{
+
+    bool setAddCurrentWorkingDirectorytoSearchPath(bool addCurrentWorkingDirectorytoSearchPath);
+
+    bool setAddInputFileDirectorytoSearchPath(bool addInputFileDirectorytoSearchPath);
+
+    bool setAddepinEnvironmentVariabletoSearchPath(bool addepinEnvironmentVariabletoSearchPath);
+
+    bool addSearchPath(const openstudio::path& searchPath);
+    bool setSearchPaths(const std::vector<openstudio::path>& searchPaths);
+
+    // Convenience, forwards to the openstudio::path equivalent
+    bool setSearchPaths(const std::vector<std::string>& searchPaths);
```

Comment:
use paths. I kept the std::vector<std::string> for convenience.

I had kept the `bool addSearchPath(const std::string& searchPath);` one to begin with, but the call to `addSearchPath("string litteral")` was ambiguous since const char * converts to both std::string and fs::path, so I removed it (instead of adding a third `addSearchPath(const char*)` overload, and having to ignore it in SWIG...)

You can still do this in ruby:

```
[1] OS-build-release(main)> m = Model.new
=> #<OpenStudio::Model::Model:0x00007b0500a14d90 @__swigtype__="_p_openstudio__model__Model">
[2] OS-build-release(main)> p = m.getPythonPluginSearchPaths
=> #<OpenStudio::Model::PythonPluginSearchPaths:0x00007b05005079b0 @__swigtype__="_p_openstudio__model__PythonPluginSearchPaths">
[3] OS-build-release(main)> p.addSearchPath("lol")
=> true
[4] OS-build-release(main)> puts p
OS:PythonPlugin:SearchPaths,
  {85828d9f-b7df-44f7-89c4-cbd18c92badc}, !- Handle
  Yes,                                    !- Add Current Working Directory to Search Path
  Yes,                                    !- Add Input File Directory to Search Path
  Yes,                                    !- Add epin Environment Variable to Search Path
  lol;                                    !- Search Path 1
```

Reply:
Python:

```python
In [1]: m = openstudio.model.Model()

In [2]: p = m.getPythonPluginSearchPaths()

In [3]: p.addSearchPath(Path('.').absolute())
Out[3]: True

In [4]: p.addSearchPath('the path')
Out[4]: True

In [5]: print(p)
OS:PythonPlugin:SearchPaths,
  {c965de9e-6e78-4a6a-aa53-5041e1ac5809}, !- Handle
  Yes,                                    !- Add Current Working Directory to Search Path
  Yes,                                    !- Add Input File Directory to Search Path
  Yes,                                    !- Add epin Environment Variable to Search Path
  /home/julien/Software/Others/EnergyPlus-build-release, !- Search Path 1
  the path;                               !- Search Path 2
```